### ISSUE TITLE
Vimium may work well on XML view page

### ISSUE THREAD

**Issue by gdh1995** [philc/vimium#1796]
We know that on a XML page, `document.createElement("div")` gets a `<div>` with `null` style, which makes Vomnibar and LinkHints do not work. But, today I find a simple way to create normal DOM nodes and make them work well successfully.

What we should do is just to use `document.createElementNS("http://www.w3.org/1999/xhtml", tagName)`. A node created by `createElementNS` has a normal "style", and its children created by setting "innerHTML" also have good "style"s, and shadowDOMs do no harm to this.

![2](https://cloud.githubusercontent.com/assets/5547703/9486653/a586d86c-4bfc-11e5-9712-27f05a4f01a2.png)

**Comment by gdh1995** [comment]
Besides, there's another way to get a node with normal "style".
``` javascript
var html = document.importNode(document.documentElement); // get a "<html></html>"
html.innerHTML = '<div></div>';
// now, `html` and `html.firstElementChild` both have normal "style"s
```

**Comment by mrmr1993** [comment]
@gdh1995 Good stuff, this works very nicely! I've pushed a PR implementing your idea from here; hope that's okay.

If you'd rather implement it yourself, let me know and I'll close the PR :)

**Comment by gdh1995** [comment]
Besides, there's another problem (or BUG ?) about DOM nodes having `null` style:
On XML view page and XHTML page, whose content types both contain "xml", a normal `div` with a normal `.style` attribute will create children with `null` styles if:
* it belongs to a shadow DOM
* the shadow DOM has been appended to `document` DOM tree.
* we make new nodes by setting `div`'s `innerHTML`.

This problem seems to have no impact on the master version, so this is just a warning.

**Comment by gdh1995** [comment]
Closed, because #1798 has been merged.
